File: core/src/main/java/com/pluxity/station/StationLine.java
package com.pluxity.station;

import com.pluxity.global.entity.BaseEntity;
import com.pluxity.line.Line;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "station_line")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class StationLine extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "station_id")
    private Station station;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "line_id")
    private Line line;

    @Builder
    public StationLine(Station station, Line line) {
        this.station = station;
        this.line = line;
    }
}

File: core/src/main/kotlin/com/pluxity/station/StationLine.kt
package com.pluxity.station

import com.pluxity.global.entity.IdentityIdEntity
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import org.hibernate.annotations.NotFound
import org.hibernate.annotations.NotFoundAction

@Entity
@Table(name = "station_line")
class StationLine(
    @JoinColumn(name = "station_id")
    @ManyToOne(fetch = FetchType.EAGER)
    @NotFound(action = NotFoundAction.IGNORE)
    var station: Station,
    @JoinColumn(name = "line_id")
    @ManyToOne(fetch = FetchType.LAZY)
    var line: Line,
) : IdentityIdEntity()
